File: C_Assignment/Assignment_45/programAQ2.c
///////////////////////////////////////////////////////////
//
//  File name : Assignement45Question2.c
//  Description : Write a program which accept matrix and reverse the contents of each row.
//  Author : Kartik Ganesh Jare
//  Date : 13/11/25
//
///////////////////////////////////////////////////////////

/*

Input :
3 2 5 9
4 3 2 2
8 4 1 5
3 9 7 5

Output :
9 5 2 3
2 2 3 4
9 1 4 8
5 7 9 3

*/

#include <stdio.h>

void ReverseRow(int Arr[][4], int iRow, int iCol)
{
    int i = 0, start = 0, end = 0;
    int temp = 0;

    for (i = 0; i < iRow; i++)
    {
        start = 0;
        end = iCol - 1;

        while (start < end)
        {
            temp = Arr[i][start];
            Arr[i][start] = Arr[i][end];
            Arr[i][end] = temp;

            start++;
            end--;
        }
    }
}

void Display(int Arr[][4], int iRow, int iCol)
{
    int i = 0, j = 0;

    for(i = 0; i < iRow; i++)
    {
        for(j = 0; j < iCol; j++)
        {
            printf("%d\t", Arr[i][j]);
        }
        printf("\n");
    }
}

int main()
{
    int iRow = 0, iCol = 0, i = 0, j = 0;
    int Arr[4][4];

    printf("Enter number of rows :\n");
    scanf("%d", &iRow);

    printf("Enter number of columns :\n");
    scanf("%d", &iCol);

    printf("Enter matrix elements :\n");
    for (i = 0; i < iRow; i++)
    {
        for (j = 0; j < iCol; j++)
        {
            scanf("%d", &Arr[i][j]);
        }
    }

    ReverseRow(Arr, iRow, iCol);

    printf("Matrix after reversing each row:\n");
    Display(Arr, iRow, iCol);

    return 0;
}

File: Java_Assignment/Assignment_51/programAQ2.java
///////////////////////////////////////////////////////////
//
//  File name : Assignement51Question2.java
//  Description : Write a java program which accept matrix from user and swap the
//                contents of each consecutive column.
//  Author : Kartik Ganesh Jare
//  Date : 13/11/25
//  
///////////////////////////////////////////////////////////

import java.util.Scanner;

class Matrix
{
    private int Arr[][];

    public Matrix(int A,int B)
    {
        Arr = new int [A][B];
    }

    public void Accept()
    {
        System.out.println("Please enter the elements of matrix");

        Scanner sobj = new Scanner(System.in);

        int i = 0, j = 0;

        for(i = 0;i < Arr.length;i++)
        {
            for(j = 0; j< Arr[i].length;j++)
            {
                Arr[i][j] = sobj.nextInt();
            }
        }
    }

    public void Display()
    {
        System.out.println("Elements of the matrix are : ");

        int i = 0, j = 0;

        for(i = 0;i < Arr.length;i++)
        {
            for(j = 0; j< Arr[i].length;j++)
            {
                System.out.print(Arr[i][j]+"\t");
            }
            System.out.println();
        }
    }

    public void SwapRows()
    {
        int j = 0 ,i = 0, temp = 0;

        for(i = 0;i < Arr.length - 1;i+=2)
        {   
            for(j = 0;j < Arr[i].length;j++)
            {
                temp = Arr[i][j];
                Arr[i][j] = Arr[i + 1][j];
                Arr[i + 1][j] = temp;
            }
        }
    } 

    public void SwapColumns()
    {
        int j = 0 ,i = 0, temp = 0;

        for(j = 0;j < Arr[0].length - 1;j+=2)
        {   
            for(i = 0;i < Arr.length;i++)
            {
                temp = Arr[i][j];
                Arr[i][j] = Arr[i][j + 1];
                Arr[i][j + 1] = temp;
            }
        }
    } 
}

public class programAQ2
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number of rows : ");
        int iRow = sobj.nextInt();

        System.out.println("Enter number of colums : ");
        int iCol = sobj.nextInt();

        Matrix mobj = new Matrix(iRow,iCol);

        mobj.Accept();

        System.out.println("\nOriginal Matrix:");
        mobj.Display();

        mobj.SwapRows();

        System.out.println("\nMatrix after swapping rows:");
        mobj.Display();

        mobj.SwapColumns();

        System.out.println("\nMatrix after swapping Columns:");
        mobj.Display();

        sobj.close();
    }   
}